cmd/kube-agent/main.go

```diff
@@ -284,6 +287,25 @@ func main() {
 					return nil
 				},
 			},
+			{
+				Name:  "update",
+				Usage: "Watch for configuration updates and restart the agent when a change is detected",
+				Flags: flags,
+				Action: func(c *cli.Context) error {
+
+					kubeAgentMonitor := agent.NewKubeAgentMonitor(cfg,
+						agent.WithAgentNamespace("mw-agent-ns"),
+						agent.WithDaemonset("mw-kube-agent"),
+						agent.WithDeployment("mw-kube-agent"),
+						agent.WithDaemonsetConfigMap("mw-daemonset-otel-config"),
+						agent.WithDeploymentConfigMap("mw-deployment-otel-config"),
+					)
+
+					kubeAgentMonitor.SetClientSet()
+					kubeAgentMonitor.ListenForConfigChanges(context.TODO())
```

Comment:
Instead of `context.TODO()`, create a new context and pass it. Like how it is done in hostagent https://github.com/middleware-labs/mw-agent/blob/master/cmd/host-agent/main.go#L276

Reply:
Okay, using the same context from unfave/cli will help to stop the execution of code, if we terminate the command, right ?